pkg/appgw/requestroutingrules_test.go

```diff
@@ -0,0 +1,115 @@
+// -------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// --------------------------------------------------------------------------------------------
+
+package appgw
+
+import (
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var expectedRedirectID = "/subscriptions/" + testFixturesSubscription +
+	"/resourceGroups/" + testFixtureResourceGroup +
+	"/providers/Microsoft.Network/applicationGateways/" + testFixtureAppGwName +
+	"/redirectConfigurations/" + agPrefix + "sslr-" +
+	testFixturesNamespace +
+	"-" +
+	testFixturesName
+
+func makeHTTPURLPathMap() network.ApplicationGatewayURLPathMap {
+	return network.ApplicationGatewayURLPathMap{
+		Name: to.StringPtr("-path-map-name-"),
+		ApplicationGatewayURLPathMapPropertiesFormat: &network.ApplicationGatewayURLPathMapPropertiesFormat{
+			PathRules: &[]network.ApplicationGatewayPathRule{
+				{
+					ID:   to.StringPtr("-the-id-"),
+					Type: to.StringPtr("-the-type-"),
+					Etag: to.StringPtr("-the-etag-"),
+					Name: to.StringPtr("/some/path"),
+					ApplicationGatewayPathRulePropertiesFormat: &network.ApplicationGatewayPathRulePropertiesFormat{
+						BackendAddressPool:    resourceRef("--BackendAddressPool--"),
```

Comment:
on Azure side, we only allow either default redirect configuration or default backendpool/backend http setting.

Reply:
Corrected!